### PR TITLE
feat: Add Support For Low level Copy

### PR DESCRIPTION
**Pull request by bbernays**
SDK changes to expose Low Level copy functions. used in snapshot create and restore

### CODE REVIEW

File: provider/execution/storage.go [hunk 1]
```diff
@@ -23,4 +24,6 @@ type Storage interface {
 type QueryExecer interface {
 	pgxscan.Querier
 	Exec(ctx context.Context, query string, args ...interface{}) error
+	RawCopyTo(ctx context.Context, w io.Writer, sql string) error
+	RawCopyFrom(ctx context.Context, r io.Reader, sql string) error
```
Comment (roneli) [inline provider/execution/storage.go:28]: would put this in another interface called Copier, as I don't think other Storages might support this. It is always best to keep interface as smallest as possible, so people can ask for the minimal interface they require for their function.

**Review by roneli — DISMISSED** [review]

File: provider/execution/storage.go (hunk 1 above)
Comment (bbernays) [reply provider/execution/storage.go:32]: I am adding Copier to QueryExec for this implementation: https://github.com/cloudquery/cloudquery/pull/449/files

Comment (disq) [reply provider/execution/storage.go:32]: Maybe leave `Copier` by itself (don't embed it somewhere else) and cast/test the passed in QueryExecer to see if it also satisfies Copier. If not, runtime error? Also make sure PgDatabase satisfies it by doing an interface check. (var _ Copier = PgDatabase ...)

Comment (roneli) [reply provider/execution/storage.go:32]: You can add a new interface in the core that includes QueryExecer + Copier interfaces so anything given to it must satisfy both.

File: provider/execution/storage.go [hunk 2]
```diff
@@ -24,3 +25,13 @@ type QueryExecer interface {
 	pgxscan.Querier
 	Exec(ctx context.Context, query string, args ...interface{}) error
 }
+
+type Copier interface {
+	RawCopyTo(ctx context.Context, w io.Writer, sql string) error
+	RawCopyFrom(ctx context.Context, r io.Reader, sql string) error
+}
+
+type LowLevelQueryExecer interface {
```
Comment (disq) [inline provider/execution/storage.go:34]: nit: This one can be defined in core I think

**Pull request merged**